data/avro-schema/src/main/resources/avro/net/corda/data/flow/event/external/ExternalEventResponse.avsc

```diff
@@ -0,0 +1,62 @@
+{
+  "type": "record",
+  "name": "ExternalEventResponse",
+  "namespace": "net.corda.data.flow.event.external",
+  "doc": "Events from workers that are sent back to the flow worker to be consumed by flows",
+  "fields": [
+    {
+      "name": "requestId",
+      "type": "string",
+      "doc": "The request id of the external event response. Used for deduplication when processing responses."
+    },
+    {
+      "name": "chunkNumber",
+      "type": [
+        "null",
+        "int"
+      ],
+      "doc": "The current chunk number that this response contains, or `null` if the response consists of a single payload that does not need to be chunked."
+    },
+    {
+      "name": "numberOfChunks",
+      "type": [
+        "null",
+        "int"
+      ],
+      "doc": "The number of chunks that are being sent back as responses, or `null` if the response consists of a single payload that does not need to be chunked."
+    },
+    {
+      "name": "payload",
+      "type": [
+        "null",
+        "net.corda.data.persistence.EntityResponse",
+        "net.corda.data.crypto.wire.ops.flow.FlowOpsResponse"
+      ],
+      "doc": "The underlying payload that can contain domain specific information that is not serialized into the {@link ExternalEventResponse#data} property. `null` if the response represents an error and therefore does not have a payload to respond with."
+    },
+    {
+      "name": "data",
```

Comment:
What's the distinction between `payload` and `data`?

Should we use a union on the response side so we reply either with success or failure, rather than nulling some fields out some of the time? (I know we use that pattern elsewhere, I just don't like it very much!)

Reply:
Removed data.